module-network.tf

```diff
@@ -0,0 +1,142 @@
+# Copyright (c) 2017, 2023 Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+data "oci_core_vcn" "oke" {
+  count  = var.create_vcn ? 0 : 1
+  vcn_id = var.vcn_id
+}
+
+locals {
+  # Created VCN if enabled, else var.vcn_id
+  vcn_id = var.create_vcn ? one(module.vcn[*].vcn_id) : var.vcn_id
+
+  # Configured VCN CIDRs if creating, else from provided vcn_id
+  vcn_cidrs = var.create_vcn ? var.vcn_cidrs : flatten(data.oci_core_vcn.oke[*].cidr_blocks)
+
+  # Created route table if enabled, else var.ig_route_table_id
+  ig_route_table_id = var.create_vcn ? one(module.vcn[*].ig_route_id) : var.ig_route_table_id
+
+  # Created route table if enabled, else var.nat_route_table_id
+  nat_route_table_id = var.create_vcn ? one(module.vcn[*].nat_route_id) : var.nat_route_table_id
+}
+
+module "vcn" {
+  count          = var.create_vcn ? 1 : 0
+  source         = "oracle-terraform-modules/vcn/oci"
+  version        = "3.5.3"
+  compartment_id = coalesce(var.network_compartment_id, local.compartment_id)
+  defined_tags   = var.defined_tags["vcn"]
+  freeform_tags  = var.freeform_tags["vcn"]
+
+  attached_drg_id              = var.drg_id != null ? var.drg_id : (var.create_drg ? module.drg[0].drg_id : null)
+  create_internet_gateway      = !(var.load_balancers == "internal" && !var.create_bastion && var.control_plane_type == "private")
```

Comment:
set the `var.create_bastion == true` to improve readability

Reply:
Variable naming and context of logical operators should provide readability with booleans - comparison with static values is unnecessary and may even cause problems with [unexpected implementation semantics](https://medium.com/@xtg/terraform-boolean-evaluation-b866b528e90b). If there's any type ambiguity (e.g. when using map lookups) then wrapping in `tobool` is a good idea/harmless, but a bool var used directly is otherwise fine on its own.